actioncable/lib/action_cable.rb

```diff
@@ -35,7 +37,15 @@ module ActionCable
       confirmation: 'confirm_subscription'.freeze,
       rejection: 'reject_subscription'.freeze
     },
-    default_mount_path: '/cable'.freeze
+    default_mount_path: '/cable'.freeze,
+    # It is important that the unsupported protocol be last in the list,
+    # as the Websocket Driver always chooses the first protocol that matches,
+    # with the intended result that if the client sends the unsupported
+    # protcol as an option, it will be the last chosen.
```

Comment:
Yeah agreed it feels a bit fragile, which is why I was so explicit. I also just added a test to make sure that the protocol is last.

However, instead I could just add in the unsupported protocol right before we instantiate the driver (in `ActionCable::Connection::WebSocket`), which I think would accomplish what you were after when you say
> Other parts of the code might even want to stay oblivious to the fact that last protocol means "unsupported"

This would keep the placement details safe from the rest of the code.